crates/ruff/src/rules/dlint/rules/bad_exec_use.rs

```diff
@@ -6,6 +6,21 @@ use ruff_macros::{derive_message_formats, violation};
 
 use crate::checkers::ast::Checker;
 
+/// ## What it does
+/// Checks that code does not call the built-in function `exec`
+///
+/// ## Why is this bad?
+/// This function makes it far too easy to achieve arbitrary code execution, so we shouldn't
+/// support it in any context.
```

Comment:
It may be good to add some explanation about what `exec` does. The real problem here is if the string passed comes from an untrusted source (or contains parts from an untrusted source)